src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -704,9 +717,10 @@ namespace OpenRCT2::PathFinding
      *  rct2: 0x0069A997
      */
     static void PeepPathfindHeuristicSearch(
-        TileCoordsXYZ loc, const TileCoordsXYZ& goal, const Peep& peep, TileElement* currentTileElement,
-        const bool inPatrolArea, uint8_t numSteps, uint16_t* endScore, Direction testEdge, uint8_t* endJunctions,
-        TileCoordsXYZ junctionList[16], uint8_t directionList[16], TileCoordsXYZ* endXYZ, uint8_t* endSteps)
+        PathFindingState& state, TileCoordsXYZ loc, const TileCoordsXYZ& goal, const Peep& peep,
+        TileElement* currentTileElement, const bool inPatrolArea, uint8_t numSteps, uint16_t* endScore, Direction testEdge,
+        uint8_t* endJunctions, TileCoordsXYZ junctionList[16], uint8_t directionList[16], TileCoordsXYZ* endXYZ,
+        uint8_t* endSteps)
```

Comment:
At some point, we really need to cut back on the arguments for this function, or at least change how they're passed. Not for this PR, though.

Reply:
Yes, I plan to have a query struct and with designated initializers it will look pretty neat.